src/gui/shell.cpp

```diff
@@ -210,6 +207,51 @@ bool Shell::setGuiFontWide(const QString& fdesc) noexcept
 	return true;
 }
 
+void Shell::updateGuiFontRegisters() noexcept
+{
+	if (!m_attached || !m_nvim || !m_nvim->api0()) {
+		return;
+	}
+
+	// Update `set guifont=`, but only if value changes
+	MsgpackRequest* getOption{ m_nvim->api0()->vim_get_option("guifont") };
+	connect(getOption, &MsgpackRequest::finished, this, &Shell::handleGuiFontOption);
+
+	// Update `:GuiFont`, but only if value changes
+	MsgpackRequest* getVariable{ m_nvim->api0()->vim_get_var("GuiFont") };
+	connect(getVariable, &MsgpackRequest::finished, this, &Shell::handleGuiFontVariable);
+}
+
+void Shell::writeGuiFontQSettings() noexcept
+{
+	QSettings settings;
```

Comment:
Don't we need to specify arguments, as in `src/gui/shell.cpp`? Actually the font was not restored in the next startup in my environment without this change.
```suggestion
	QSettings settings { "nvim-qt", "nvim-qt" };
```

Reply:
Yes, currently.

Ideally, I'd like to add some `QSettings` test coverage. This is a byproduct of some soon-to-be-added changes, the new call structure is necessary for a test-mock.

Side note: How did you get this cool "Commit suggestion" dialog?

Reply:
> Side note: How did you get this cool "Commit suggestion" dialog?

I think this is very useful feature too :)
Usage is very simple. When you add a comment to the code as usual, press the icon indicated below.

![image](https://user-images.githubusercontent.com/20490597/125108234-be1f7a80-e11c-11eb-9b02-e0a202b9637b.png)

Then the commenting line will be automatically inserted as a code block of type `suggestion`.

![image](https://user-images.githubusercontent.com/20490597/125108324-e0b19380-e11c-11eb-8e32-7133fb270d5d.png)

You can make a suggestion by editing the inserted code block.

![image](https://user-images.githubusercontent.com/20490597/125108455-076fca00-e11d-11eb-9ebe-fb53a517e5cc.png)

Finally you will get the following dialog.

![image](https://user-images.githubusercontent.com/20490597/125108635-44d45780-e11d-11eb-8bd1-309ff7a35719.png)